src/compile.js

```diff
@@ -1259,15 +1266,136 @@ Compiler.prototype.ctryexcept = function (s) {
 
     this.setBlock(orelse);
     this.vseqstmt(s.orelse);
-    // Should jump to finally, but finally is not implemented yet
     this._jump(end);
+
     this.setBlock(end);
 };
 
+Compiler.prototype.outputFinallyCascade = function (thisFinally) {
+    var nextFinally;
+    
+    if (this.u.finallyBlocks.length == 0) {
+        out("if($postfinally!==undefined) { if ($postfinally.returning) { return $postfinally.returning; } else { $blk=$postfinally.gotoBlock; $postfinally=undefined; continue; } }");
+    } else {
+        // If this was a nonlocal return, we need to work out where to go.
+        // If it's a return, we always continue the finally chain.
+        // If it's a break or continue, we need to work out whether our parent
+        // finally-block is at the same break/continue depth as us (ie whether there's
+        // a loop/breakable-block boundary between us and it).
+
+        nextFinally = this.peekFinallyBlock();
+        
+        out("if($postfinally!==undefined) {",
```

Comment:
This is the best I've got: https://docs.python.org/2.7/reference/compound_stmts.html#finally

Relevant section:
> When a return, break or continue statement is executed in the try suite of a try...finally statement, the finally clause is also executed ‘on the way out.’

The challenge is executing a `break` or `continue` in a nested `finally` block. What I'm doing here is working out whether the  *which* `finally` blocks need to execute for a given `break` (or `continue`). I compare the `breakDepth` of the inner and outer `try` blocks (that is, how many nested `break`able blocks surround them). If the number is the same, then the `break` will always go through both `finally` blocks. If they're different, we only need to run the inner one before resuming.

(Aside: Looks like you're right - `break` and `continue` contexts are always exactly the same. They're tracked separately in `compile.js`, so I had a brainfart and just assumed Python had a `switch` statement or something else `break`able but not `continue`able. So I suppose we could get away with only tracking one of them.)